### DISCUSSION TITLE
Derived atom with dynamic path and uncertain depth

### DISCUSSION THREAD

**Discussion by xd-hearst** [pmndrs/jotai#2213] [Q&A] [answered]
I am trying to change my `selectAtom` to just use a derived atom, since it is the recommended way, https://github.com/pmndrs/jotai/discussions/2209#discussioncomment-7380257

Though the path of the derived atom could be quite dynamic and of uncertain depth. 

Right now i use `selectAtom` in the following way, 

```
const complicatedAtom = atom({ data: { a: { b: { c: 1, d: [{ x: 2 }] } } } });

// from https://github.com/developit/dlv/blob/master/index.js
function dlv(obj, key, def, p, undef) {
  key = key.split ? key.split(".") : key;
  for (p = 0; p < key.length; p++) {
    obj = obj ? obj[key[p]] : undef;
  }
  return obj === undef ? def : obj;
}

function useStableAtomValue(key, baseAtom) {
  const myAtom = useMemo(
    () =>
      selectAtom(baseAtom, (p) => {
        return dlv(p, key);
      }),
    [key, baseAtom]
  );

  return useAtomValue(myAtom);
}
```

So is it possible to not use `selectAtom` in this case? A sample sandbox can be found here: https://codesandbox.io/s/jolly-currying-pck552

**Comment by dai-shi** [comment, marked as answer]
```diff
  const myAtom = useMemo(
    () =>
-     selectAtom(baseAtom, (p) => {
+     atom((get) => {
+       const p = get(baseAtom)
        return dlv(p, key);
      }),
    [key, baseAtom]
  );
```

**Reply by xd-hearst** [reply]
oh, thank you so much. 
Cannot overstate how much we appreciate the great libraries you created and all the help you have generously provided to the community.  

One little follow up question, what is the advantage of this over `selectAtom`?  I thought and was hoping that with `get` we can avoid using `useMemo`.

**Reply by dai-shi** [reply]
See #2210 

No, you can't avoid useMemo in either case, because your atom depends on props.
